### PR TITLE
chore: convert zoekenservice to Kotlin

### PR DESCRIPTION
**Pull request by edgarvonk**
Converted zoekenservice from Java to Kotlin for better maintainability. Added integration test and unit test.

Solves PZ-4428

### CODE REVIEW

File: src/itest/kotlin/nl/lifely/zac/itest/ZaakRestServiceTest.kt [hunk 1]
```diff
@@ -106,7 +107,7 @@ class ZaakRestServiceTest : BehaviorSpec({
                       "isProcesGestuurd": false,
                       "isVerlengd": false,
                       "kenmerken": [],
-                      "omschrijving": "dummyOmschrijving",
+                      "omschrijving": "$ZAAK_DESCRIPTION_2",
```
Comment (bas-info-nl) [inline src/itest/kotlin/nl/lifely/zac/itest/ZaakRestServiceTest.kt:110]: There's no need to put a string `ZAAK_DESCRIPTION_2` in a string through parameter injection. Or am I missing something here? And this is done in a number of places, I'd expect a detekt warning would've mentioned this, no?

File: src/main/kotlin/net/atos/zac/solr/SolrUtil.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+private const val SOLR_ESCAPE = '\\'
+private const val SOLR_QUOTE = '\"'
+
+/**
+ * Produces a quoted Solr string, with properly encoded contents, from a raw Java string.
+ *
+ * @param value the raw unencoded string
+ * @return the encoded and quoted Solr string
+ */
+fun quoted(value: String) = SOLR_QUOTE.toString() + encoded(value) + SOLR_QUOTE
```
Comment (bas-info-nl) [inline src/main/kotlin/net/atos/zac/solr/SolrUtil.kt:17]: Is this not easier?
```suggestion
fun quoted(value: String) = "$SOLR_QUOTE${encoded(value)}$SOLR_QUOTE"
```

Comment (bas-info-nl) [reply src/main/kotlin/net/atos/zac/solr/SolrUtil.kt:17]: Or even as an extension function?

**Review by bas-info-nl — APPROVED** [review]

File: src/main/kotlin/net/atos/zac/solr/SolrUtil.kt (hunk 2 above)
Comment (edgarvonk) [reply src/main/kotlin/net/atos/zac/solr/SolrUtil.kt]: Used your suggestion, thanks! Extension function is also nice yes. Maybe another day. :-)

**Pull request merged**